docs/bsp.md

```diff
@@ -502,7 +525,118 @@ trait WorkspaceBuildTargetsResult {
 }
 ```
 
-### 1.6.3. Build Target Changed Notification
+### Register File Watcher Request
+
+The file watching request is sent from the server to the client to notify that the server is
+interested in watching file changes coming from the client (the editor, or a language server
+forwarding `workspace/didChangeWatchedFiles`). These file changes can be used to invalidate tasks
+(like compilation) or caches in the build tool.
+
+For a motivation of this feature, check [the LSP `workspace/didChangedWatchedFiles`
+description](https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles).
+
+Request:
+
+* method: `build/registerFileWatcher`
+* params: `RegisterFileWatcherParams`, defined as follows
+
+```scala
+trait RegisterFileWatcherParams {
+   /** The glob pattern to watch in all the workspace.
+     * Syntax is implementation specific. */
```

Comment:
Let's try to make an effort to define the syntax of this magic string 

> The glob pattern to watch in the workspace, for example `src/**/*.{scala,js}`.
> - `*` Matches 0 or more characters in a single path portion
> - `**` matches zero or more path portions
> - `*.{scala,js}` expands to two patterns: `*.scala` and `*.js`
> - Forward slashes `/` are always treated as path separators, even on Windows.
>
> Further details are implementation defined.

Reply:
Sounds reasonable to me :+1: Pity that LSP doesn't define on its own.